### ISSUE TITLE
NVDA-alpha-23765 abnormal behavior on the secure desktop

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#12852]
### Steps to reproduce:
1. Install Alpha-23765.
2. Press Ctrl + Alt + Delete to enter the safe desktop;
3. Press the Tab key to try to navigate the contents of the secure desktop.
4. NVDA+N to open the NVDA menu.

### Actual behavior:
1. Execute step 3 NVDA does not announce anything.
2. Execute step 4 NVDA can announce the contents of the NVDA menu normally.

### Expected behavior:
Execute step 3, NVDA reads "lock", "logout" and other items.

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
Alpha-23765 
#### Windows version:
Windows 10 21H1 (x64) build 19043.1237
#### Name and version of other software in use when reproducing the issue:
None
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA-Alpha-23763 and NVDA 2021.2 work as expected
#### If add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by leonardder** [comment]
I can reproduce this.

**Comment by leonardder** [comment]
Filed #12854

**Comment by lukaszgo1** [comment]
While #12854 probably fixes this specific case it does not address the underlying issue.
The log snipped for this issue is as follows
```

  File "core.pyc", line 237, in _setInitialFocus
  File "NVDAObjects\__init__.pyc", line 301, in objectWithFocus
  File "NVDAObjects\__init__.pyc", line 231, in findBestAPIClass
  File "NVDAObjects\__init__.pyc", line 226, in findBestAPIClass
  File "NVDAObjects\window\__init__.pyc", line 104, in getPossibleAPIClasses
  File "_UIAHandler.pyc", line 818, in isUIAWindow
  File "_UIAHandler.pyc", line 757, in _isUIAWindowHelper
  File "baseObject.pyc", line 26, in __get__
  File "appModuleHandler.pyc", line 413, in _get_productName
  File "appModuleHandler.pyc", line 405, in _setProductInfo
  File "appModuleHandler.pyc", line 363, in _getExecutableFileInfo
  File "fileUtils.pyc", line 48, in getFileVersionInfo
RuntimeError: The file C:\Windows\System32\consent.exe does not exist

```

This occurs because when checking if C:\Windows\System32\consent.exe exists NVDA is redirected to syswow64 as we are a 32-bit executable. I went with a different approach which ensures that for the duration of retrieving file information wow64 redirection is disabled https://github.com/lukaszgo1/nvda/commit/d7a7d4b5616880953ac3ffb322e9e8630c453cc0) and confirmed it works with a self signed  build. Obviously for a real PR I'll tidy this code up so wow64redirection is disabled only if the file path is from System32 but it shows the general idea. @leonardder  What do you think?

**Comment by leonardder** [comment]
Thanks for investigating @lukaszgo1! I think we should go for both approaches as https://github.com/nvaccess/nvda/pull/12854 improves the logic regardless, as it ensures that the code checking the productName and version only runs when the class is wwg, and therefore in far fewer cases. I've also seen an application failing outside the secure desktop that had no version information available, so the issue we're dealing with is broader than secure desktop and your patch wouldn't fix that.

**Comment by lukaszgo1** [comment]
> This occurs because when checking if C:\Windows\System32\consent.exe exists NVDA is redirected to syswow64 as we are a 32-bit executable. I went with a different approach which ensures that for the duration of retrieving file information wow64 redirection is disabled [lukaszgo1@d7a7d4b](https://github.com/lukaszgo1/nvda/commit/d7a7d4b5616880953ac3ffb322e9e8630c453cc0)) and confirmed it works with a self signed build. Obviously for a real PR I'll tidy this code up so wow64redirection is disabled only if the file path is from System32 but it shows the general idea. 

It turns out that  32-bit systems need to be handled differently when disabling WOW64 redirection and I don't dare to submit a PR without real life testing. This would need to wait until I'll have a time to create a 32-bit VM  and test there.
